autofaiss/indices/index_utils.py

```diff
@@ -155,3 +156,11 @@ def _download_one(src_dst_path: Tuple[str, str], fs: fsspec.AbstractFileSystem):
     with ThreadPool(min(16, len(indices_file_paths))) as pool:
         for _ in pool.imap_unordered(partial(_download_one, fs=fs), src_dest_paths):
             pass
+
+
+def initialize_direct_map(index: faiss.Index) -> None:
+    nested_index = extract_index_ivf(index) if isinstance(index, faiss.swigfaiss.IndexPreTransform) else index
```

Comment:
This condition is not sufficient, it is possible to wrap other things below an index pre-transform (even if on the subset of indices we support it works)

2 ways of solving this issue would be:
- to do a try catch
- To parse the index_key of the index and check if it contains IVF